### PR TITLE
Add helpful hints to `VGroup.add()` error message

### PR DESCRIPTION
**Pull request by vvolhejn**
## Overview: What does this pull request change?

Changes the error message to explicitly state what was wrong and suggest a solution.

## Motivation and Explanation: Why and how do your changes improve the library?

I was confused by the current error message when I was creating a large `VGroup`. The solution is to use `Group` instead as I discovered [here](https://github.com/3b1b/manim/issues/795), but I think that's not obvious to somebody not familiar with Manim internals.

## Further Information and Comments

This is tested by `test_vectorized_mobject.py::test_vgroup_init`.

There are about a dozen other places where error messages could be improved similarly - search for "must be of type" in the codebase. I didn't want to change all of them at once in case there are reviewer objections, but I'm happy to change other similar messages too.

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

**Review by chopan050 — APPROVED** [review]
Looks good to me! This kind of changes is very needed and appreciated. Thanks for your contribution!

File: manim/mobject/types/vectorized_mobject.py [hunk 1]
```diff
@@ -2003,8 +2003,14 @@ def construct(self):
                         (gr-circle_red).animate.shift(RIGHT)
                     )
         """
-        if not all(isinstance(m, (VMobject, OpenGLVMobject)) for m in vmobjects):
-            raise TypeError("All submobjects must be of type VMobject")
+        for m in vmobjects:
+            if not isinstance(m, (VMobject, OpenGLVMobject)):
+                raise TypeError(
+                    f"All submobjects of {self.__class__.__name__} must be of type VMobject. "
+                    f"Got {repr(m)} ({type(m)}) instead. "
```
Comment (JasonGrace2282) [inline manim/mobject/types/vectorized_mobject.py:2010]:
```suggestion
                    f"Got {repr(m)} ({type(m).__name__}) instead. "
```
Maybe slightly better looking?

**Review by JasonGrace2282 — APPROVED** [review]
Overall nice change!

File: manim/mobject/types/vectorized_mobject.py (hunk 1 above)
Comment (vvolhejn) [reply manim/mobject/types/vectorized_mobject.py:2010]: Works for me. `type()` gives the full path, which is more explicit. But then again, we are already using `__name__` for the other type, and I've seen it in other places in the codebase.

**Pull request merged**